### ISSUE TITLE
Data race in rule_applicator.go

### ISSUE THREAD

**Issue by rspurgeon** [daveshanley/vacuum#487]
When using vacuum (`github.com/daveshanley/vacuum v0.9.14`) in a integration test library, we receive the following DATA RACE errors. 

```
=== RUN   Test_LintPlain
=== RUN   Test_LintPlain/lint_plain
==================
WARNING: DATA RACE
Read at 0x00c000558630 by goroutine 51:
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet.func7()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:597 +0x274
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet.gowrap1()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:615 +0x54

Previous write at 0x00c000558630 by goroutine 52:
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet.func7()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:598 +0x28c
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet.gowrap1()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:615 +0x54

Goroutine 51 (running) created at:
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:566 +0x5d8c
  github.com/kong/deck/cmd.executeLint()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/cmd/file_lint.go:113 +0x214
  github.com/spf13/cobra.(*Command).execute()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xbc4
  github.com/spf13/cobra.(*Command).ExecuteC()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x4b8
  github.com/spf13/cobra.(*Command).Execute()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039 +0x204
  github.com/spf13/cobra.(*Command).ExecuteContext()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032 +0x200
  github.com/kong/deck/tests/integration.lint()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/tests/integration/test_utils.go:338 +0x1c4
  github.com/kong/deck/tests/integration.Test_LintPlain.func1()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/tests/integration/lint_test.go:31 +0xd4
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40

Goroutine 52 (running) created at:
  github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/daveshanley/vacuum@v0.9.14/motor/rule_applicator.go:566 +0x5d8c
  github.com/kong/deck/cmd.executeLint()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/cmd/file_lint.go:113 +0x214
  github.com/spf13/cobra.(*Command).execute()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xbc4
  github.com/spf13/cobra.(*Command).ExecuteC()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x4b8
  github.com/spf13/cobra.(*Command).Execute()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039 +0x204
  github.com/spf13/cobra.(*Command).ExecuteContext()
      /Users/rick.spurgeon@konghq.com/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032 +0x200
  github.com/kong/deck/tests/integration.lint()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/tests/integration/test_utils.go:338 +0x1c4
  github.com/kong/deck/tests/integration.Test_LintPlain.func1()
      /Users/rick.spurgeon@konghq.com/go/src/github.com/Kong/deck/tests/integration/lint_test.go:31 +0xd4
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40
==================
    testing.go:1398: race detected during execution of test
--- FAIL: Test_LintPlain (0.02s)
    --- FAIL: Test_LintPlain/lint_plain (0.02s)
FAIL
FAIL    github.com/kong/deck/tests/integration  0.894s
FAIL
```

Looking for any tips or resolution. Thank you

**Comment by daveshanley** [comment]
strange, will look into it for you - I don't know what the issue is at the moment.

**Comment by daveshanley** [comment]
There are a few different cases causing this, should be fixed in the next release.

**Comment by daveshanley** [comment]
This has been resolved in `v0.9.15`

**Comment by rspurgeon** [comment]
Thank you @daveshanley !  I will upgrade and report any issues. Very appreciated.
